MarkBind/markbind
# Update all references of deprecated URL `parse` to use `parse` from a new dependency `url-parse`

**What is the purpose of this pull request?**

- [X] Others, please explain: Updating of deprecated API



**Overview of changes:**

Fixes #1944.

Previously, we imported `url` module from `node` in-built modules and used `url.parse(...)` to create a `URL` object. However, it has been deprecated.

Now, we introduce a new dependency, `url-parse`, and its type defintions i.e. `@types`. This dependency builds on Node.js `url` interface/related API.

Helpful links:
- [url-parse GitHub repo](https://github.com/unshiftio/url-parse)
- [@types/url-parse npm page](https://www.npmjs.com/package/@types/url-parse)

**Anything you'd like to highlight/discuss:**

NIL

**Testing instructions:**

I believe we just have to build the backend, `npm run build:backend`, and run `markbind s -d` inside the `docs/` folder to ensure that the page loads successfully.

Next, `npm run test` should fully pass.



**Proposed commit message: (wrap lines at 72 characters)**


Replace deprecated url parse with url-parse dependency

---

**Checklist:** :ballot_box_with_check:
